tests/periph_uart_mode/main.c

```diff
@@ -0,0 +1,181 @@
+#define MODE_STR_LEN (4)
+
+/* UART_DEV is always 0 since this test is for device with 1 UART */
+#define _UART_DEV (UART_DEV(0))
+
+
+/* Delay between each mode, makes parsing results easier */
+#define DELAY_US 50000
+
+
+static void _get_mode(uart_data_bits_t data_bits, uart_parity_t parity,
+                        uart_stop_bits_t stop_bits, char* mode_str) {
+    switch (data_bits) {
+        case UART_DATA_BITS_5:
+            mode_str[DATA_BIT_POS] = '5';
+            break;
+        case UART_DATA_BITS_6:
+            mode_str[DATA_BIT_POS] = '6';
+            break;
+        case UART_DATA_BITS_7:
+            mode_str[DATA_BIT_POS] = '7';
+            break;
+        case UART_DATA_BITS_8:
+            mode_str[DATA_BIT_POS] = '8';
+            break;
+        default:
+            break;
+    }
+
+    switch (parity) {
+        case UART_PARITY_NONE:
+            mode_str[PARITY_POS] = 'N';
+            break;
+        case UART_PARITY_EVEN:
+            mode_str[PARITY_POS] = 'E';
+            break;
+        case UART_PARITY_ODD:
+            mode_str[PARITY_POS] = 'O';
+            break;
+        case UART_PARITY_MARK:
+            mode_str[PARITY_POS] = 'M';
+            break;
+        case UART_PARITY_SPACE:
+            mode_str[PARITY_POS] = 'S';
+            break;
+        default:
+            break;
+    }
+
+    switch (stop_bits) {
+        case UART_STOP_BITS_1:
+            mode_str[STOP_BIT_POS] = '1';
+            break;
+        case UART_STOP_BITS_2:
+            mode_str[STOP_BIT_POS] = '2';
+            break;
+        default:
+            break;
+    }
+
+    mode_str[MODE_STR_LEN - 1] = '\0';
+}
+
+
+int main(void)
+{
+    int8_t status;
+
+    /* Stores each mode string for printing at the end of the test */
+    char mode_strings[TOTAL_OPTIONS][MODE_STR_LEN];
+
+    /* Stores the result for each mode */
+    bool results[TOTAL_OPTIONS];
+
+    /* Results index that is used to associate mode_strings and results */
+    uint8_t ridx = 0;
+
+    /* Use arrays so that each option can be easily iterated over */
+    uart_data_bits_t data_bits[DATA_BIT_OPTIONS] = {
+        UART_DATA_BITS_5,
+        UART_DATA_BITS_6,
+        UART_DATA_BITS_7,
+        UART_DATA_BITS_8
+    };
+
+    uart_parity_t parity[PARITY_OPTIONS] = {
+        UART_PARITY_NONE,
+        UART_PARITY_EVEN,
+        UART_PARITY_ODD,
+        UART_PARITY_MARK,
+        UART_PARITY_SPACE
+    };
+
+    uart_stop_bits_t stop_bits[STOP_BIT_OPTIONS] = {
+        UART_STOP_BITS_1,
+        UART_STOP_BITS_2
+    };
+
+    /* Test each permutation */
+    for (int didx = 0; didx < 4; ++didx) {
+        for (int pidx = 0; pidx < 5; ++pidx) {
+            for (int sidx = 0; sidx < 2; ++sidx) {
+                /* Initialize the UART before each mode switch */
+                uart_init(_UART_DEV, STDIO_UART_BAUDRATE, NULL, NULL);
+                /* Initialize stdio to get printf */
+                stdio_init();
```

Comment:
This seems redundant now, I think either uart init or stdio init will work.

Reply:
Done